File: kotlin/grade-school/src/main/kotlin/GradeSchool.kt
class School {
    private val students = mutableMapOf<Int, MutableList<String>>()
    fun add(student: String, grade: Int) {
        students.getOrPut(grade) { mutableListOf() }.add(student); students[grade]!!.sort()
    }

    fun grade(grade: Int) = students.getOrPut(grade) { mutableListOf() }
    fun roster() = students.toSortedMap().flatMap { it.value }
}

File: ruby/grade-school/grade_school.rb
class School
  def initialize
    @school = Hash.new { [] }
  end

  def add(student, level)
    @school[level] <<= student
  end

  def students(x)
    @school[x].sort
  end

  def students_by_grade
    @school.keys.map { { grade: _1, students: students(_1) } }.sort_by { _1[:grade] }
  end
end
